server/api-util/currency.js

```diff
@@ -0,0 +1,179 @@
+const Decimal = require('decimal.js');
+const has = require('lodash/has');
+const { types } = require('sharetribe-flex-sdk');
+const { Money } = types;
+
+/** Helper functions for handling currency */
+
+// https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+// https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Number/MIN_SAFE_INTEGER
+// https://stackoverflow.com/questions/26380364/why-is-number-max-safe-integer-9-007-199-254-740-991-and-not-9-007-199-254-740-9
+const MIN_SAFE_INTEGER = Number.MIN_SAFE_INTEGER || -1 * (2 ** 53 - 1);
+const MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 2 ** 53 - 1;
+
+const isSafeNumber = decimalValue => {
+  if (!(decimalValue instanceof Decimal)) {
+    throw new Error('Value must be a Decimal');
+  }
+  return decimalValue.gte(MIN_SAFE_INTEGER) && decimalValue.lte(MAX_SAFE_INTEGER);
+};
+
+// See: https://en.wikipedia.org/wiki/ISO_4217
+// See: https://stripe.com/docs/currencies
+const subUnitDivisors = {
+  AUD: 100,
+  CAD: 100,
+  CHF: 100,
+  CNY: 100,
+  DKK: 100,
+  EUR: 100,
+  GBP: 100,
+  HKD: 100,
+  INR: 100,
+  JPY: 1,
+  MXN: 100,
+  NOK: 100,
+  NZD: 100,
+  SEK: 100,
+  SGD: 100,
+  USD: 100,
+};
+
+// Get the minor unit divisor for the given currency
+exports.unitDivisor = currency => {
+  if (!has(subUnitDivisors, currency)) {
+    throw new Error(`No minor unit divisor defined for currency: ${currency}`);
+  }
+  return subUnitDivisors[currency];
+};
+
+////////// Currency manipulation in string format //////////
+
+/**
+ * Ensures that the given string uses only dots or commas
+ * e.g. ensureSeparator('9999999,99', false) // => '9999999.99'
+ *
+ * @param {String} str - string to be formatted
+ *
+ * @return {String} converted string
+ */
+const ensureSeparator = (str, useComma = false) => {
```

Comment:
These "ensureSeparator" stuff were added due to users writing listing price in different localized number rules.
Is it likely that these server-side tests would get random input too? 
(These probably don't hurt the test, but they just obfuscate the code a bit.)

Reply:
We are not expecting random input because the value comes always form Money object so I guess there is no need to test this.